app/server/utils/transfer_limits.py

```diff
@@ -0,0 +1,251 @@
+
+from server.models import token
+from server.models.credit_transfer import CreditTransfer
+from server.utils.transfer_enums import TransferSubTypeEnum, TransferTypeEnum
+
+AppliedToTypes = List[Union[TransferTypeEnum,
+                            Tuple[TransferTypeEnum, TransferSubTypeEnum]]]
+
+PAYMENT = TransferTypeEnum.PAYMENT
+DEPOSIT = TransferTypeEnum.DEPOSIT
+WITHDRAWAL = TransferTypeEnum.WITHDRAWAL
+
+AGENT_OUT = TransferSubTypeEnum.AGENT_OUT
+STANDARD = TransferSubTypeEnum.STANDARD
+
+STANDARD_PAYMENT = (PAYMENT, STANDARD)
+AGENT_OUT_PAYMENT = (PAYMENT, AGENT_OUT)
+
+
+def get_transfer_limits(credit_transfer: CreditTransfer):
+    relevant_limits = []
+    for limit in LIMITS:
+        applied = limit.application_filter(credit_transfer)
+        if applied and (credit_transfer.transfer_type in limit.applied_to_transfer_types
+                        or (credit_transfer.transfer_type, credit_transfer.transfer_subtype)
+                        in limit.applied_to_transfer_types):
+            
+            relevant_limits.append(limit)
+
+    return relevant_limits
+
+
+    # # Supports filtering over type-subtype tuples of the form ('PAYMENT', 'AGENT_OUT')
+    # applied = limit.application_filter(self)
+    # if applied:
+    #     for transfer_type in limit.applied_to_transfer_types:
+    #         if isinstance(transfer_type, (tuple, list)):
+    #             if str(self.transfer_type) == transfer_type[0]\
+    #                     and str(self.transfer_subtype) == transfer_type[1]:
+    #                 relevant_limits.append(limit)
+    #                 continue
+    #         else:
+    #             if str(self.transfer_type) == transfer_type:
+    #                 relevant_limits.append(limit)
+    #                 continue
+
+# ~~~~~~SIMPLE CHECKS~~~~~~
+
+
+def user_has_group_account_role(credit_transfer):
+    return credit_transfer.sender_user.has_group_account_role
+
+
+def user_phone_is_verified(credit_transfer):
+    return credit_transfer.sender_user.is_phone_verified
+
+
+def user_individual_kyc_is_verified(credit_transfer):
+    if credit_transfer.sender_user is not None:
+        kyc = credit_transfer.sender_user.kyc_applications
+        if len(kyc) > 0 and kyc[0].kyc_status == 'VERIFIED' and kyc[0].type == 'INDIVIDUAL':
+            return True
+
+    return False
+
+
+def user_business_kyc_is_verified(credit_transfer):
+    if credit_transfer.sender_user is not None:
+        kyc = credit_transfer.sender_user.kyc_applications
+        if len(kyc) > 0 and kyc[0].kyc_status == 'VERIFIED' and kyc[0].type == 'BUSINESS':
+            return True
+
+    return False
+
+
+def token_is_liquid_type(credit_transfer):
+    return credit_transfer.token and credit_transfer.token.token_type is token.TokenType.LIQUID
+
+
+def transfer_is_agent_out_subtype(credit_transfer):
+    return credit_transfer.transfer_subtype is TransferSubTypeEnum.AGENT_OUT
+
+# ~~~~~~COMPOSITE CHECKS~~~~~~
+
+
+def is_liquid_and_has_foo(ct):
```

Comment:
:thinking_face:

Reply:
ohhhh shit that was from when i was talking to u about SLAP during that call lol! Can u remove when u change the implementation of sempo limits?